examples/core_api/timm-post/post.md

```diff
@@ -0,0 +1,286 @@
+
+To follow along, all you'll need is the Determined CLI and a Determined cluster with at least 2 GPUs.  Check out https://docs.determined.ai/latest/getting-started.html for details.  Scripts and configuration files for this post can be downloaded at (TODO).
+
+# Step 0: Running the Script
+To start off, we'll show how to run the script on a single GPU with no modifications.  We copied the script unchanged from https://github.com/rwightman/pytorch-image-models/blob/master/train.py to `step0-run.py`.  Changes in 0.17.XX allow us to run arbitrary executables in a Determined experiment by setting them as our  `entrypoint`, so we can use the following configuration to train on CIFAR10:
+```name: step0-run.yaml
+entrypoint: python step0-run.py --dataset=torch/cifar10 --dataset-download data --input-size 3 32 32 --epochs 5
+max_restarts: 0
+searcher:
+   name: single
+   max_length: 1
+   metric: val_loss
+```
+We'll talk more about `searcher` in Step 4, but for now we can safely ignore `max_length` and `metric` .  Setting the searcher to `single` means this experiment will create only one trial.
+
+We also need to make sure `timm` gets installed in each task container.  We'll use the following `startup-hook.sh` file, which will run in each container before running our `entrypoint`:
+```name: startup-hook.sh
+pip install timm
+```
+
+We can now launch training using `det experiment create step0.yaml .` and monitor training logs through the Web UI.
+
+# Step 1: Distributed Training
+
+The `timm` training script uses PyTorch Distributed to do distributed data parallel training, with the essential bit of configuration being this call:
+```
+torch.distributed.init_process_group(backend="nccl", init_method="env://")
+```
+[init_process_group](https://pytorch.org/docs/stable/distributed.html#torch.distributed.init_process_group) will then pull information from environment variables about how to connect with other processes and coordinate distributed training.
+
+When we set `slots_per_trial` to something bigger than 1 in our experiment configuration, the Determined master will take care of starting up multiple copies of our entrypoint in containers distributed appropriately across our cluster -- we just need to set those environment variables correctly in each container.  In 0.17.XX, we've provided a [launch script](https://github.com/determined-ai/determined/tree/master/harness/determined/launch) that wraps an entrypoint and handles this environment setup, along with similar scripts for DeepSpeed and Horovod.  Our main step toward distributed training is to modify our `entrypoint` with the appropriate launch script.  While we're at it, we'll bump `slots_per_trial` to 2 to tell the Determined master we want to use 2 GPUs:
```

Comment:
Gene has requested that we stick to the phase "launcher" instead of "launch layer" or "launch script", which I agree with, so we should probably change that throughout.